### ISSUE TITLE
rename `twisted.web2.iweb.IOldRequest` to `twisted.web.iweb.IRequest`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3416]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3416|
|Type|enhancement|
|Created|2008-09-07 12:23:59Z|
|Branch|https://github.com/twisted/twisted/tree/iresource-3416-2|

`twisted.web` has been missing `IRequest` for quite a while.  Apparently it is almost already defined in `twisted.web2`.  The definition belongs in `twisted.web`, though.

<details><summary>Searchable metadata</summary>

```
trac-id__3416 3416
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_iresource_3416_2 branches/iresource-3416-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1220790239000000 1220790239000000
changetime__1231438733000000 1231438733000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24761]) Branching to 'iresource-3416'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Moved

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

What about @since tags? Or am I missing something here..

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Good catch, thanks. :)  Hopefully after you correct me on that two or three more times I'll start to remember on my own.

This is a bit tricky, since the interface itself is new, but it documents something that's old, except some of the attributes are actually also new.  So I guess I'm going to do the safe thing and just say it's since 8.2.

Thanks for the review (not sure if that was meant to be a whole review or if you just wanted to point out that one thing, so putting it back up for review).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_4):
> Good catch, thanks. :)  Hopefully after you correct me on that two or three more times I'll start to remember on my own.

Yep thats the plan :)

> Thanks for the review (not sure if that was meant to be a whole review or if you just wanted to point out that one thing, so putting it back up for review).

Yeah, just wanted to point out that thing. Another thing though, while working on [[#1852](https://github.com/twisted/twisted/issues/1852)](https://github.com/twisted/twisted/issues/1852) I noticed some methods in this interface lack docstrings and have redundant pass statements. Can you fix that or is #1852 a more appropriate ticket for that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [25068]) Branching to 'iresource-3416-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

[#3273](https://github.com/twisted/twisted/issues/3273) was a duplicate of this ticket.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner|
|-|-|

This seems to be ready for another review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set owner to @exarkun|
|-|-|

It looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [25888]) Merge iresource-3416-2

Author: exarkun
Reviewer: thijs, therve
Fixes: [#3416](https://github.com/twisted/twisted/issues/3416)

Add `twisted.web.iweb.IRequest` defining the interface provided by request objects
in Twisted Web.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
